velox/functions/sparksql/String.h

```diff
@@ -583,4 +583,83 @@ struct OverlayVarbinaryFunction : public OverlayFunctionBase {
   }
 };
 
+/// translate(string, matchStr, replaceStr) -> varchar
+///
+///   Returns a new translated string. It translates the character in ``string``
+///   by a character in ``replaceStr``. The character in ``replaceStr`` is
+///   corresponding to the character in ``matchStr``. The translation will
+///   happen when any character in ``string`` matching with the character in the
+///   ``matchStr``.
+template <typename T>
+struct TranslateFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  // ASCII input always produces ASCII result.
+  static constexpr bool is_default_ascii_behavior = true;
+
+  using MapType = std::unordered_map<StringView, StringView>;
+  MapType dict_;
+  bool isConstantTranslateDict = false;
+
+  FOLLY_ALWAYS_INLINE MapType buildDict(
+      const arg_type<Varchar>& matchStr,
+      const arg_type<Varchar>& replaceStr) {
+    MapType dict;
+    int i = 0, j = 0;
+    while (i < matchStr.size()) {
+      StringView replacePiece;
+      if (j >= replaceStr.size()) {
+        replacePiece = StringView("");
```

Comment:
This logic suggests that 'replace' string may be shorter than 'match' string, in which case remaining characters in 'match' are removed from the input string. Is this how Spark's function behaves? If so, please, document this behavior. It would be nice to add some examples to documentation to illustrate common cases and special cases.

Reply:
Yes, this is exactly spark's behavior. Just documented. Thank!